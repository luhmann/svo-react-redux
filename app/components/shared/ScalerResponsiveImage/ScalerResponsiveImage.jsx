import React from 'react'
import { buildScalerImageUrl } from 'lib/UrlBuilder.js'

import {RESPONSIVE_IMAGE_SIZES} from '../../../constants/Common.js'

const buildSrcSet = (objectId, sizes) => {
  return sizes
    .map((size) => (
      `${buildScalerImageUrl(objectId, size)} ${size}w`
    ))
    .join(', ')
}

const ScalerResponsiveImage = ({ img, className = '', srcset = RESPONSIVE_IMAGE_SIZES, sizes = '100vw' }) => (
  <img
    className={className}
    src={buildScalerImageUrl(img.objectId)}
    srcSet={buildSrcSet(img.objectId, srcset)}
    sizes={sizes} />
)

ScalerResponsiveImage.propTypes = {
  img: React.PropTypes.shape({
    objectId: React.PropTypes.string.isRequired,
    credits: React.PropTypes.string
  }),
  className: React.PropTypes.string,
  srcset: React.PropTypes.array,
  sizes: React.PropTypes.string
}

export default ScalerResponsiveImage
