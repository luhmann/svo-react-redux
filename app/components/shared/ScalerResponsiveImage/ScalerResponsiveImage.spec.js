import React from 'react'
import renderer from 'react-test-renderer'

import ScalerResponsiveImage from './ScalerResponsiveImage.jsx'

describe.only('Component: <ScalerResponsiveImage />', () => {
  it('should render a responsive image', () => {
    const img = {
      objectId: 'lh1nmgmqugmlopswifqt.jpg',
      credits: 'Jan Florian Dietrich'
    }

    const component = renderer.create(
      <ScalerResponsiveImage img={img} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
