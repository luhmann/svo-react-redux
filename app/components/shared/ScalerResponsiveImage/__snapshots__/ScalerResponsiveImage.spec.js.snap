exports[`Component: <ScalerResponsiveImage /> should render a responsive image 1`] = `
<img
  className=""
  sizes="100vw"
  src="http://svo.dev:9000/lh1nmgmqugmlopswifqt.jpg"
  srcSet="http://svo.dev:9000/lh1nmgmqugmlopswifqt.jpg/:/rs=w:414/fx-sp 414w, http://svo.dev:9000/lh1nmgmqugmlopswifqt.jpg/:/rs=w:1024/fx-sp 1024w, http://svo.dev:9000/lh1nmgmqugmlopswifqt.jpg/:/rs=w:1280/fx-sp 1280w, http://svo.dev:9000/lh1nmgmqugmlopswifqt.jpg/:/rs=w:1366/fx-sp 1366w, http://svo.dev:9000/lh1nmgmqugmlopswifqt.jpg/:/rs=w:1920/fx-sp 1920w, http://svo.dev:9000/lh1nmgmqugmlopswifqt.jpg/:/rs=w:2500/fx-sp 2500w" />
`;
